client/my-sites/site-settings/podcasting-details/index.jsx

```diff
@@ -349,7 +358,23 @@ class PodcastingDetails extends Component {
 	};
 
 	onCategoryCleared = () => {
-		this.props.updateFields( { podcasting_category_id: '0' } );
+		const {
+			updateFields,
+			clearDirtyFields,
+			submitForm,
+			siteSlug,
+			isSaveRequestSuccessful,
+		} = this.props;
+
+		updateFields( { podcasting_category_id: '0' }, () => {
+			submitForm();
+		} );
+
+		if ( isSaveRequestSuccessful ) {
+			// Save changed fields and allow redirect
+			clearDirtyFields();
+			page.redirect( '/settings/writing/' + siteSlug );
+		}
```

Comment:
This doesn't look quite right.

- How will `onCategoryCleared` be called again after the successful save request?
- Which fields are still dirty after saving?

I would have expected to see something like this instead:

```js
updateFields( { ... }, () => {
    submitForm( ( err ) => { // not real code, the point is to execute something after `submitForm` finishes
        if ( ! err ) {
            page.redirect( ... );
        }
    } );
} );